business-networks-workflows/src/main/kotlin/net/corda/bn/flows/BNService.kt

```diff
@@ -283,6 +284,28 @@ class BNService(private val serviceHub: AppServiceHub) : SingletonSerializeAsTok
         }
     }
 
+    /**
+     * Checks if a set of permissions can be removed from the network (by changing roles of a member, suspending or revoking members).
+     *
+     * @param networkId ID of the Business Network.
+     *
+     * @param permissions Set of administrative permissions which will be removed from the network as a consequence of one of the following operations:
+     * suspension, revocation, role change
+     *
+     * @return [true] if each permission in the provided set occurs on more than one member in the network, [false] otherwise.
+     */
+    fun safeToRemovePermissions(networkId: String, permissions: Set<AdminPermission>): Boolean {
+        val existingPermissions = getAllMemberships(networkId).map { membership ->
+            membership.state.data.roles.map { it.permissions }.flatten()
+        }.flatten().filterIsInstance<AdminPermission>()
+
+        permissions.forEach { permission ->
+            if (existingPermissions.count { it ==  permission } <= 1)
```

Comment:
nit: Extra whitespace:
```suggestion
            if (existingPermissions.count { it == permission } <= 1)
```

Reply:
Done.